lib/file.mjs

```diff
@@ -313,26 +310,48 @@ class File extends EventEmitter {
       })
 
       const getChunk = () => {
-        if (stopped) return
-        const chunkOffset = currentOffset
-        const chunkMax = Math.min(end, chunkOffset + chunkSize - 1)
-        if (chunkMax < chunkOffset) {
+        if (currentOffset > end) {
           stopped = true
           return
         }
 
-        const writeStream = new PassThrough()
-        appendStream(null, writeStream)
+        const chunkOffset = currentOffset
+        const chunkMax = Math.min(end, chunkOffset + chunkSize - 1)
+        const chunkNumber = lastStartedChunk++
 
         let tries = 0
         const tryFetchChunk = () => {
           tries++
+
           this.api.fetch(response.g + '/' + chunkOffset + '-' + chunkMax)
             .then(response => {
               handleMegaErrors(response)
               return response.arrayBuffer()
             }).then(data => {
-              writeStream.end(Buffer.from(data))
+              const dataBuffer = Buffer.from(data)
+
+              // Check if this chunk is the next in the steam
+              if (nextChunk === chunkNumber) {
+                // If it is then write to the stream
+                decryptStream.write(dataBuffer)
+
+                // Check if the next chunk is in the buffer
+                while (true) {
+                  const bufferChunk = chunkBuffer[++nextChunk]
+                  if (!bufferChunk) break
+                  decryptStream.write(dataBuffer)
```

Comment:
Should have been `bufferChunk`. Anyway, I will implement backpressure handling too.